### ISSUE TITLE
Provide support for a multilingual plugin

### ISSUE THREAD

**Issue by ajmas** [github/pages-gem#401]
Some pages have a need to be multilingual, so it would be useful to have support for one of the multilingual Jekyll plugins in the gh-pages whitelist, without having to resort to using a CI for this.

Two possibilities:

   - [jekyll-multiple-languages-plugin](https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin) 
   - [jekyll-multilingual](https://github.com/sylvaindurand/jekyll-multilingual)

**Comment by untra** [comment]
Another possibility
* [polyglot](polyglot.untra.io)
that my plugin :abc: 

I made an issue here asking what I need to do to get it supported : https://github.com/github/pages-gem/issues/438

**Comment by benbalter** [comment]
I'd love to add multi-lingual support. It looks like there are multiple multi-lingual plugins out there. Is there a clear front runner? One with clear advantages over others?

**Comment by nschonni** [comment]
Going by Gem downloads it would be https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin
The only possible problem with the Polyglot one, might be how it overrides the build https://github.com/untra/polyglot#how-it-works

**Comment by benbalter** [comment]
> Going by Gem downloads it would be https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin

I'd be hesitant to add a plugin without a test suite to ensure the intended functionality. Adding a plugin to millions of sites will naturally expose edge cases, feature requests, and contributions, and we'd want to make sure that (positive) attention can be incorporate into a healthy project without risking the existing functionality, etc.

**Comment by DirtyF** [comment]
@benbalter What do you think about asking explicitly on the ISSUE template that before suggesting a plugin for GitHub Pages, it should be available as a gem, covered with a test suite?

**Comment by benbalter** [comment]
@DirtyF Great call! Will get that going (unless you'd like to?)

**Comment by DirtyF** [comment]
@benbalter I did a proposal in #473.

**Comment by yatil** [comment]
Without having an answer for the best multi language plugin, I just want to add my support to adding one. It would best support both use cases:

- Build a complete alternative website for another language (e.g. /en/ for English, /es/ for Spanish, …) and
- Support the translation of any document (for example having a blog and several pages in English but also having about-me.md in a German and English version)

Thanks for looking into this!

**Comment by dora-gt** [comment]
Although considering security and risk is important, but please imagine that there are so many customers that can't serve multiple-language pages. I really need the function.

**Comment by DirtyF** [comment]
GitHub Pages comes with limitations.

Fortunately, it's possible to host a Jekyll repository on GitHub and easily overcome those limitations by linking your GitHub repo to [Netlify](https://www.netlify.com/) for instance. Netlify will automatically build your custom Jekyll site - with any plugin you require - when you push a commit to your GitHub Branch.

**Comment by dora-gt** [comment]
Thank you for the suggestion. I don't know whether the options are suitable for my case, anyway, I look into it. Thank you.

**Comment by larsrc** [comment]
For those who can't wait on a plugin solution, https://www.sylvaindurand.org/making-jekyll-multilingual/ has a plugin-free solution which AFAICT would lend itself to an eventual switch to a plugin.

**Comment by 88kbbq** [comment]
I've tried jekyll-multiple-languages-plugin but it breaks everything. Can't find support, will try [Iarsrc link](https://www.sylvaindurand.org/making-jekyll-multilingual/)

**Comment by joepio** [comment]
I'm using jekyll-multiple-languages-plugin as well, and ended up compiling everything locally using the `gh-pages` npm package and running `gh-pages -d _site` after every change, to commit and push to a `gh-pages` branch.

**Comment by erotavlas85** [comment]
I found this [how to](https://www.klaasnotfound.com/2017/02/16/proper-multilingual-site-with-github-pages-and-jekyll/) about  [jekyll-multiple-languages-plugin](https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin).
In your opinion what is the best solution to follow jekyll-multiple-languages-plugin or [jekyll-multilingual](https://www.sylvaindurand.org/making-jekyll-multilingual/) without plugin?

**Comment by 88kbbq** [comment]
Use data files for site specific language data and font matter for page specific data. I’ll post a temple link for regency when I’m back.

**Comment by DirtyF** [comment]
ICYMI [jekyll-locale](https://github.com/ashmaroli/jekyll-locale) is a plugin is currently in development to help handle i18n in Jekyll.

**Comment by augnustin** [comment]
Up on the topic!

Where are we now? What can be done? 

👍 for https://github.com/Anthony-Gaudino/jekyll-multiple-languages-plugin

**Comment by alr-lab** [comment]
So to brainstorm the issue discussions, the only thing blocking Github Pages to whitelist any Jekyll language plugin is the lack of tests?

**Comment by pbatard** [comment]
It's a bit saddening to see that this issue has been opened for 4 years, still without any sign of an upcoming resolution.

I do understand that validating a plugin, and ensuring that it does not contribute to introducing a global security risk, is tricky, and does indeed warrant extra caution, but, considering that GitHub's new parent company, Microsoft, has always been at the forefront of providing localised content as well as easing up localisation matters for developers, it's disappointing not to see GitHub embrace the same stance and make sure that people, who do use GitHub's recommended services to provide web pages, can make these pages easily accessible for people of all horizons.

Thus, in case GitHub staff needs a little more convincing that this is an important matter, let me give you a concrete example of how the lack of progress on this issue is negatively impacting some GitHub users. In my case, as the developer of [Rufus](https://github.com/pbatard/rufus), which is a well a rather popular Windows Open Source utility, I am currently using GitHub pages to host the web site, and I must commend GitHub for making this service available to its users, on account that it has greatly reduced the headaches I was having with custom server hosting. As [you can see however](https://rufus.ie) the web page for the application is quite plain looking, and I sure wouldn't mind being able to modernize it, as well as make its maintenance less dependent on editing plain HTML, by switching to GitHub's attractive Jekyll static page generation...

However, because of this very issue, I just cannot do that, because I also must provide localised versions of my content, and, if I am going to move to new ways of generating it, I sure would like to take this opportunity to also do away with the [current mess](https://github.com/pbatard/rufus-web) of having to:
1. Locally host a [php version of the page](https://github.com/pbatard/rufus-web/blob/gh-pages/index.php) on a server that can serve dynamic content (so that I can use `php-gettext` to generate localized pages).
2. Run a [local script](https://github.com/pbatard/rufus-web/blob/gh-pages/_refresh.sh) to re-generate the 56 localised static pages every time I update content.
3. Push these 56 regenerated localised static pages to GitHub.

Thus I genuinely can't wait for the day where I finally switch to Jekyll generated content, with a GitHub supported multilingual plugin, and end this brittle, hard to support & hard to upgrade content generation process (which I'm sure many other people, who need to provide localised versions of GitHub pages have their own different version of). But for that to happen, it looks like GitHub will need to show the same level of commitment to localisation that Microsoft has, and bite the bullet on investing resources to test and validate at least one Jekyll multilingual plugin...

Hopefully then, after more than 4 years of users **patiently** waiting for progress on this matter, GitHub staff might consider that the benefits of solving this issue for users of GitHub services worldwide, does outweight the cost of having to approve a new Gem, and may finally see this issue come to some form of resolution.

Thank you.

**Comment by BaseMax** [comment]
Hi, I used **"jekyll-multiple-languages-plugin"** plugin in a website and now want to serve that in GitHub Pages, But seems it's not possible.

There is no plan to whitelist this plugin?
No idea? No, decide? no plan?
It's not a new issue!
https://pages.github.com/versions/

Thank you

**Comment by augnustin** [comment]
I guess this is due to the fact that those american based companies think English is the only language around. 

What a pity. Such cultural impoverishment.

**Comment by ajmas** [comment]
In many ways the Jekyll team should be choosing a multilingual solution and pushing it on the community. 

I ended choosing Hugo for my pages, since multilingual support is part of its architecture. I was also working on a project on GitLab, where they are more agnostic, but they have CI templates to deal with them. 

Maybe we just need some GitHub workflow templates and then just go towards static website generators that are multilingual friendly?

**Comment by vijaya-lakshmi-venkatraman** [comment]
Hi
Is there any update on this plugin?

**Comment by pbatard** [comment]
Well, considering that there still hasn't been any progress in 7 years, I will ask that GitHub either closes this issue altogether or provides us with an ETA/outline of what they are planning to do about it. It is not acceptable to leave your users in limbo about this __for more than 7 years__.

So I would request that GitHub makes a formal decision on whether they are planning to ever add support for a multilingual plugin, and have the courtesy to post it here. Thank you.
